rskj-core/src/main/java/co/rsk/peg/Bridge.java

```diff
@@ -617,15 +618,13 @@ public void addSignature(Object[] args) throws VMException {
             }
             signatures.add(signatureByteArray);
         }
-        byte[] rskTxHash = (byte[]) args[2];
-        if (rskTxHash.length!=32) {
-            throw new BridgeIllegalArgumentException("Invalid rsk tx hash " + Bytes.of(rskTxHash));
-        }
+        byte[] rskTxHashSerialized = (byte[]) args[2];
         try {
+            Keccak256 rskTxHash = deserializeRskTxHash(rskTxHashSerialized);
             bridgeSupport.addSignature(federatorPublicKey, signatures, rskTxHash);
-        } catch (BridgeIllegalArgumentException e) {
-            throw e;
-        } catch (Exception e) {
+        } catch (IllegalArgumentException e) {
+            throw new BridgeIllegalArgumentException("Invalid rsk tx hash " + Bytes.of(rskTxHashSerialized));
+        } catch (IOException e) {
```

Comment:
Is this change a good idea? I think all Bridge methods are capturing generic exceptions

Reply:
There might be unchecked exception thrown

Reply:
updated